src/testdrive/src/main.rs

```diff
@@ -187,6 +194,29 @@ async fn main() {
         args.max_errors
     );
 
+    let arg_vars: Result<BTreeMap<String, String>, String> = args
+        .arg_var
+        .iter()
+        .map(|arg| {
+            let mut parts = arg.splitn(2, '=');
+            let name = parts
+                .next()
+                .ok_or_else(|| "No key for --arg-var".to_string())?;
+            validate_var_name(name)?;
```

Comment:
Can you push the `validate_var_name` check into the loop in `action::build`? That way `validate_var_name` doesn't need to be a weird part of the public API. Unlike many of our crates, `testdrive`-the-library actually has a very refined public API: https://dev.materialize.com/api/rust/testdrive/index.html

Reply:
Yeah, originally that was where I put this, but then it felt like doing this one bit of argument validation so far away from all the other arg validation was weird.